tests/lib/nrf_modem_lib/nrf91_sockets/src/nrf91_sockets_test.c

```diff
@@ -0,0 +1,1932 @@
+	__wrap_nrf_freeaddrinfo_Stub(nrf_freeaddrinfo_stub);
+
+	ret = getaddrinfo(HTTPS_HOSTNAME, NULL, &hints, &res);
+
+	TEST_ASSERT_EQUAL(ret, DNS_EAI_FAMILY);
+}
+
+void test_nrf91_socket_offload_getaddrinfo_ipv4_success(void)
+{
+	int ret;
+	struct addrinfo *res;
+	struct addrinfo hints = {
+		.ai_family = AF_INET,
+		.ai_socktype = SOCK_STREAM,
+		.ai_canonname = "name",
+	};
+	struct nrf_sockaddr_in info_in = { 0 };
+
+	test_state_nrf_getaddrinfo.res.ai_family = NRF_AF_INET;
+	test_state_nrf_getaddrinfo.res.ai_socktype = NRF_SOCK_STREAM;
+	test_state_nrf_getaddrinfo.res.ai_protocol = NRF_IPPROTO_TCP;
+	test_state_nrf_getaddrinfo.res.ai_addr = (struct nrf_sockaddr *)&info_in;
+	test_state_nrf_getaddrinfo.ret = 0;
+
+	__wrap_nrf_getaddrinfo_Stub(nrf_getaddrinfo_stub);
+	__wrap_nrf_freeaddrinfo_Stub(nrf_freeaddrinfo_stub);
+
+	ret = getaddrinfo(HTTPS_HOSTNAME, NULL, &hints, &res);
+	TEST_ASSERT_EQUAL(ret, 0);
+}
+
+void test_nrf91_socket_offload_getaddrinfo_ipv6_success(void)
+{
+	int ret;
+	struct addrinfo *res;
+	struct addrinfo hints = {
+		.ai_family = AF_INET6,
+		.ai_socktype = SOCK_STREAM,
+	};
+	struct nrf_sockaddr_in6 info_in = { 0 };
+
+	test_state_nrf_getaddrinfo.res.ai_family = NRF_AF_INET6;
+	test_state_nrf_getaddrinfo.res.ai_socktype = NRF_SOCK_STREAM;
+	test_state_nrf_getaddrinfo.res.ai_protocol = NRF_IPPROTO_TCP;
+	test_state_nrf_getaddrinfo.res.ai_addr = (struct nrf_sockaddr *)&info_in;
+	test_state_nrf_getaddrinfo.ret = 0;
+
+	__wrap_nrf_getaddrinfo_Stub(nrf_getaddrinfo_stub);
+	__wrap_nrf_freeaddrinfo_Stub(nrf_freeaddrinfo_stub);
+
+	ret = getaddrinfo(HTTPS_HOSTNAME, NULL, &hints, &res);
+
+	TEST_ASSERT_EQUAL(ret, 0);
+}
+
+void test_nrf91_socket_offload_close_wrong_socket(void)
+{
+	int ret;
+
+	ret = close(-1);
+
+	TEST_ASSERT_EQUAL(ret, -1);
+	TEST_ASSERT_EQUAL(errno, EBADF);
+}
+
+void test_nrf91_socket_offload_create_close_success(void)
+{
+	int ret;
+	int fd;
+	int nrf_fd = 2;
+	int family = AF_INET;
+	int type = SOCK_STREAM;
+	int proto = IPPROTO_TCP;
+
+	__wrap_nrf_socket_ExpectAndReturn(NRF_AF_INET, NRF_SOCK_STREAM,
+					  NRF_IPPROTO_TCP, nrf_fd);
+
+	fd = socket(family, type, proto);
+
+	TEST_ASSERT_EQUAL(fd, 0);
+
+	__wrap_nrf_close_ExpectAndReturn(nrf_fd, 0);
+
+	ret = close(fd);
+
+	TEST_ASSERT_EQUAL(ret, 0);
+}
+
+void test_nrf91_socket_offload_create_close_fail(void)
+{
+	int fd;
+	int family = AF_INET;
+	int type = SOCK_STREAM;
+	int proto = IPPROTO_TCP;
+
+	__wrap_nrf_socket_ExpectAndReturn(NRF_AF_INET, NRF_SOCK_STREAM,
+					  NRF_IPPROTO_TCP, -1);
+
+	fd = socket(family, type, proto);
+
+	TEST_ASSERT_EQUAL(fd, -1);
+}
+
+void test_nrf91_socket_offload_create_native_close_fail(void)
+{
+	int fd;
+	int family = AF_INET;
+	int type = SOCK_STREAM | SOCK_NATIVE;
+	int proto = IPPROTO_TCP;
+
+	fd = socket(family, type, proto);
+
+	TEST_ASSERT_EQUAL(errno, EAFNOSUPPORT);
+	TEST_ASSERT_EQUAL(fd, -1);
+}
+
+void test_nrf91_socket_offload_create_native_tls_close_fail(void)
```

Comment:
Why is close in the test name here?

Reply:
I'm not to fund of `fail` or `error` in test names as this is something I search for in CI to find issues.
Maybe call it `test_nrf91_socket_offload_create_native_tls_eafnosupport`?

Reply:
I removed all occurrences of `fail` in test names, where possible updated with error code, where not possible I changed it to `error`. At the moment I don't see how else to name these particular cases.